3party/opening_hours/opening_hours.cpp

```diff
@@ -748,4 +749,40 @@ bool OpeningHours::IsValid() const
 {
   return m_valid;
 }
+bool OpeningHours::IsTwentyFourHours() const
+{
+  return m_rule.size() == 1 && m_rule[0].IsTwentyFourHours();
+}
+
+bool OpeningHours::HasWeekdaySelector() const
+{
+  return std::any_of(begin(m_rule), end(m_rule), [](RuleSequence const & seq)
+                     {
+                       return seq.HasWeekdays();
+                     });
+}
+
+bool OpeningHours::HasMonthSelector() const
+{
+  return std::any_of(begin(m_rule), end(m_rule), [](RuleSequence const & seq)
+                     {
+                       return seq.HasMonths();
+                     });
+}
+
+bool OpeningHours::HasWeekSelector() const
+{
+  return std::any_of(begin(m_rule), end(m_rule), [](RuleSequence const & seq)
```

Comment:
mem_fun is deprecated, use mem_fn
http://en.cppreference.com/w/cpp/utility/functional/mem_fun